contracts/v2/self-service/SelfServiceEditionCurationV4.sol

```diff
@@ -0,0 +1,298 @@
+    address _optionalSplitAddress,
+    uint256 _optionalSplitRate,
+    uint256 _totalAvailable,
+    uint256 _priceInWei,
+    uint256 _startDate,
+    uint256 _endDate,
+    uint256 _artistCommission,
+    string _tokenUri
+  )
+  public
+  onlyOwner
+  returns (uint256 _editionNumber)
+  {
+    require(_artistCommission.add(_optionalSplitRate).add(koCommission) <= 100, "Total commission exceeds 100");
+
+    uint256 editionNumber = _createEdition(_artist, _enableAuction, _totalAvailable, _priceInWei, _artistCommission, _tokenUri);
+
+    if (_startDate > 0) {
+      kodaV2.updateStartDate(editionNumber, _startDate);
+    }
+
+    if (_endDate > 0) {
+      kodaV2.updateEndDate(editionNumber, _endDate);
+    }
+
+    if (_optionalSplitRate > 0) {
+      kodaV2.updateOptionalCommission(_editionNumber, _optionalSplitRate, _optionalSplitAddress);
+    }
+
+    frequencyControls.recordSuccessfulMint(_artist, _totalAvailable, _priceInWei);
+
+    return editionNumber;
+  }
+
+  /**
+   * @dev Internal function for edition creation
+   */
+  function _createEdition(
+    address _artist,
+    bool _enableAuction,
+    uint256 _totalAvailable,
+    uint256 _priceInWei,
+    uint256 _artistCommission,
+    string _tokenUri
+  )
+  internal
+  returns (uint256 _editionNumber){
+
+    // Enforce edition size
+    require(_totalAvailable > 0, "Must be at least one available in edition");
+    require(_totalAvailable <= maxEditionSize, "Must not exceed max edition size");
+
+    // Enforce min price
+    require(_priceInWei >= minPricePerEdition, "Price must be greater than minimum");
+
+    // If we are the owner, skip this artists check
+    if (msg.sender != owner) {
+      // Enforce who can call this
+      require(accessControls.isEnabledForAccount(_artist), "Only allowed artists can create editions for now");
+    }
+
+    // Find the next edition number we can use
+    uint256 editionNumber = getNextAvailableEditionNumber();
+
+    // Attempt to create a new edition
+    require(
+      _createNewEdition(editionNumber, _artist, _totalAvailable, _priceInWei, _artistCommission, _tokenUri),
+      "Failed to create new edition"
+    );
+
+    // Enable the auction if desired
+    if (_enableAuction) {
+      auction.setArtistsControlAddressAndEnabledEdition(editionNumber, _artist);
+    }
+
+    // Trigger event
+    emit SelfServiceEditionCreated(editionNumber, _artist, _priceInWei, _totalAvailable, _enableAuction);
+
+    return editionNumber;
+  }
+
+  /**
+   * @dev Internal function for calling external create methods with some none configurable defaults
+   */
+  function _createNewEdition(
+    uint256 _editionNumber,
+    address _artist,
+    uint256 _totalAvailable,
+    uint256 _priceInWei,
+    uint256 _artistCommission,
+    string _tokenUri
+  )
+  internal
+  returns (bool) {
+    return kodaV2.createActiveEdition(
+      _editionNumber,
+      0x0, // _editionData - no edition data
+      1, // _editionType - KODA always type 1
```

Comment:
feels like we will start using `type` more. Pass in?

Reply:
Good idea